donate/payments/views.py

```diff
@@ -329,12 +329,27 @@ def process_monthly_transaction(self, form):
             return self.process_braintree_error_result(result, form)
 
     def get_transaction_details_for_session(self, result, form, **kwargs):
+        card_type = kwargs.get('card_type', 'Unknown')
```

Comment:
It doesn't look like `card_type` is every being passed to the kwargs? This needs to be supplied in the calls to `self.success()` that happen in `process_single_transaction` and `process_monthly_transaction`.

Reply:
It's not, because this is the function where card_type, as one of the transaction details, gets extracted from Braintree's `result` transaction.  The only reason this line is `kwargs.get('card_type', 'Unknown')` instead of `= 'Unknown'` is because for some reason, two tests for "monthly" put their own `card_type` in `kwargs` (something I don't see happening during real monthly upsell user actions) and so if this function does not check whether something has put `card_type` in `kwargs`, the payload and confirmation don't agree on `card_type` in the two tests mentioned in the next line.